### PR TITLE
 PI-1125 - Refactor and Add Supplier Assessment Appointment Rescheduling Tests

### PR DESCRIPTION
**Pull request by smalepati99**
This pull request introduces Playwright tests to validate the rescheduling functionality of Supplier Assessment Appointments. The tests encompass different scenarios, including rescheduling to a future date (refactoring), rescheduling to a past date with attendance and notification settings, and verifying the appointments in the Delius system.

The key modifications made in this pull request are as follows:

1. Integration of Playwright tests to cover the rescheduling of Supplier Assessment Appointments.
2. Refactoring of the verifyContact function in find-contacts.js to enhance code structure and introduce an optional parameter, isNSIContact, for handling NSI Contacts. This enhancement improves the function's flexibility and readability.
3. Optimization of test functions by eliminating unnecessary code and enhancing code structure.
4. Refactoring of helper functions to improve readability and reusability.

These tests ensure the accurate functioning of the rescheduling feature and proper synchronization of appointments in the Delius system.

### CODE REVIEW

File: steps/delius/contact/find-contacts.ts [hunk 1]
```diff
@@ -119,14 +86,44 @@ export const rescheduleSupplierAssessmentAppointment = async (
...
+    }
+
+    // return to progress screen
+    await page.click('a.govuk-button')
+    await expect(page).toHaveURL(/service-provider\/referrals\/.*\/progress/)
+
     const rescheduledAppointmentDateTime = await page
         .locator('[data-cy="supplier-assessment-table"] .govuk-table__cell:first-child')
         .innerText()
+    console.log(rescheduledAppointmentDateTime)
```
Comment (anthony-britton-moj) [inline steps/delius/contact/find-contacts.ts]: do we need this console.log??

File: tests/refer-and-monitor-and-delius/reschedule-appointment.spec.ts [hunk 2]
```diff
@@ -40,12 +39,18 @@ test('Reschedule Supplier Assessment Appointment to future date', async ({ page
     // Verify the initial Supplier Assessment Appointment in Delius
     await loginDelius(page)
     await navigateToNSIContactDetails(page, crn)
-    await verifyNSIContactInDelius(
+    const formattedInitialDateTime = formatRMDateToDelius(initialAppointmentDateTime)
+    console.log('************INITIAL FORMATTED DATE******************' + formattedInitialDateTime)
```
Comment (anthony-britton-moj) [inline tests/refer-and-monitor-and-delius/reschedule-appointment.spec.ts]: we don't really want this logging do we?

File: steps/delius/contact/find-contacts.ts (hunk 1 above)
Comment (smalepati99) [reply steps/delius/contact/find-contacts.ts]: Removed it

File: tests/refer-and-monitor-and-delius/reschedule-appointment.spec.ts (hunk 2 above)
Comment (smalepati99) [reply tests/refer-and-monitor-and-delius/reschedule-appointment.spec.ts]: Removed it

File: steps/referandmonitor/appointment.ts [hunk 3]
```diff
@@ -110,7 +107,7 @@ const editSession = async (page: Page, referralRef: string, detail: SessionDetai
     }
 
     await page.waitForURL(
-        /service-provider\/action-plan\/.*\/appointment\/.*\/post-session-feedback\/edit\/.*\/check-your-answers/
+        /service-provider\/referrals\/.*\/supplier-assessment\/post-assessment-feedback\/edit\/.*\/check-your-answers/
```
Comment (anthony-britton-moj) [inline steps/referandmonitor/appointment.ts]: can we check this please ... this is `edit session` ?? so needs to be post-session-feedback still

**Review by anthony-britton-moj — CHANGES_REQUESTED** [review]

**Review by anthony-britton-moj — DISMISSED** [review]

**Review by anthony-britton-moj — DISMISSED** [review]

**Review by anthony-britton-moj — APPROVED** [review]

**Pull request merged**